src/routes/download/index.ts

```diff
@@ -477,4 +478,56 @@ router.get('/file/:form/:blob', async (req, res) => {
   });
 });
 
+router.post('/export/pdf/', async (req, res) => {
+  try {
+    const authToken: any = req.headers.authorization.split('Bearer ')[1];
+    const accessTokenId: any = req.headers.access_token;
+
+    const browser = await puppeteer.launch({
+      executablePath: process.env.CHROME_BIN || null,
+      headless: true,
+      //args: ['--no-sandbox', '--disable-setuid-sandbox','--disable-web-security', '--enable-features=NetworkService'],
+      args: ['--no-sandbox'],
+      ignoreDefaultArgs: ['--disable-extensions'],
+    });
+    const page = await browser.newPage();
+
+    await page.evaluateOnNewDocument(
+      async (idToken, accessToken) => {
+        localStorage.clear();
+        localStorage.setItem('idtoken', idToken);
+        localStorage.setItem('access_token', accessToken);
+        localStorage.setItem('id_token', idToken);
+      },
+      authToken,
+      accessTokenId
+    );
+    await page.goto(req.body.dashboard_url, { waitUntil: 'networkidle0' });
+    const cookies = [
+      {
+        name: 'apollo-server-landing-page-redirect-to-studio-local',
+        value: 'false',
+      },
+    ];
+
+    await page.setCookie(...cookies);
+    const fileName = (Math.random() + 1).toString(36).substring(2);
+    console.log('fileName ==>> ', fileName);
+    const filePath = `/home/node/app/pdf/${fileName}.pdf`;
+    await page.pdf({
```

Comment:
@ketanreliefapps what would be nice to  have is to be able to determine if the download is really successful
Indeed, unauthorirez user could try to download a pdf of à dashboard it does not have accès to
How to deal with that ?
Is there anything using puppeteer that could tell that the user was not redirected, or could check that the html contains for example specific éléments ( html id for example ) that could identify the content ?

Reply:
@AntoineRelief , can you please give me unique class name or id for check page redirect on dashboard or not?